### DISCUSSION TITLE
How to correctly use shap.maskers.Fixed()？

### DISCUSSION THREAD

**Discussion by ribatsua** [shap/shap#3222] [Q&A]
I couldn't find a demo for shap.maskers.Fixed(). When I try to use it, I always encounter the following error while calculating shap_values: IndexError: index 0 is out of bounds for axis 1 with size 0.
![image](https://github.com/shap/shap/assets/97381556/fab429d0-8923-41a0-adca-1ba83068950b)

----> 1 shap_values = explainer(Test_data[1:2])
from tensorflow.keras.applications.resnet50 import ResNet50, preprocess_input
import json
import shap
import tensorflow as tf

# load pre-trained model and choose two images to explain
model = ResNet50(weights='imagenet')
def f(X):
    tmp = X.copy()
    preprocess_input(tmp)
    return model(tmp)
X, y = shap.datasets.imagenet50()

# load the ImageNet class names as a vectorized mapping function from ids to names
url = "https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json"
with open(shap.datasets.cache(url)) as file:
    class_names = [v[1] for v in json.load(file).values()]

# define a masker that is used to mask out partitions of the input image, this one uses a blurred background
#masker = shap.maskers.Image("inpaint_telea", X[0].shape)
masker = shap.maskers.Fixed()

# By default the Partition explainer is used for all  partition explainer
explainer = shap.Explainer(f, masker, output_names=class_names)

# here we use 500 evaluations of the underlying model to estimate the SHAP values
shap_values = explainer(X[1:3], max_evals=500, batch_size=50, outputs=shap.Explanation.argsort.flip[:1])
shap.image_plot(shap_values)

Encountered the same error across different models and datasets.

D:\conda\envs\TF23\lib\site-packages\shap\explainers\_permutation.py in __call__(self, max_evals, main_effects, error_bounds, batch_size, outputs, silent, *args)
     82         return super().__call__(
     83             *args, max_evals=max_evals, main_effects=main_effects, error_bounds=error_bounds, batch_size=batch_size,
---> 84             outputs=outputs, silent=silent
     85         )
     86 

D:\conda\envs\TF23\lib\site-packages\shap\explainers\_explainer.py in __call__(self, max_evals, main_effects, error_bounds, batch_size, outputs, silent, *args, **kwargs)
    266             row_result = self.explain_row(
    267                 *row_args, max_evals=max_evals, main_effects=main_effects, error_bounds=error_bounds,
--> 268                 batch_size=batch_size, outputs=outputs, silent=silent, **kwargs
    269             )
    270             values.append(row_result.get("values", None))

D:\conda\envs\TF23\lib\site-packages\shap\explainers\_permutation.py in explain_row(self, max_evals, main_effects, error_bounds, batch_size, outputs, silent, *row_args)
    171         else:
    172             masks = np.zeros(1, dtype=np.int)
--> 173             outputs = fm(masks, zero_index=0, batch_size=1)
    174             expected_value = outputs[0]
    175             row_values = np.zeros((len(fm),) + outputs.shape[1:])

D:\conda\envs\TF23\lib\site-packages\shap\utils\_masked_model.py in __call__(self, masks, zero_index, batch_size)
     61             else:
     62                 full_masks = np.zeros((int(np.sum(masks >= 0)), self._masker_cols), dtype=np.bool)
---> 63                 _convert_delta_mask_to_full(masks, full_masks)
     64                 return self._full_masking_call(full_masks, zero_index=zero_index, batch_size=batch_size)
     65 

D:\conda\envs\TF23\lib\site-packages\shap\utils\_masked_model.py in _convert_delta_mask_to_full(masks, full_masks)
    286 
    287         if masks[masks_pos] != MaskedModel.delta_mask_noop_value:
--> 288             full_masks[i,masks[masks_pos]] = ~full_masks[i,masks[masks_pos]]
    289         masks_pos += 1
    290 

IndexError: index 0 is out of bounds for axis 1 with size 0

**Comment by detrin** [comment]
My guess is that there is incorrect dimensionality in the data that you are putting into the explainer vs data that were used for training. Could you create minimal working example and create an issue for this?
